src/SeqCli/Forwarder/Channel/ForwardingChannelMap.cs

```diff
@@ -45,30 +48,65 @@ ForwardingChannel OpenOrCreateChannel(string? apiKey, string name)
             apiKey,
             _shutdownTokenSource.Token);
     }
-
-    public ForwardingChannel Get(string? apiKey)
+    
+    void ReopenApiKeyChannels()
     {
-        if (string.IsNullOrWhiteSpace(apiKey))
+        if (_config.Forwarder.UseApiKeyForwarding)
         {
-            return _defaultChannel;
+            foreach (var directoryPath in Directory.EnumerateDirectories(_bufferPath))
+            {
+                if (directoryPath.Equals(GetStorePath("Default"))) continue;
+                
+                var path = new SystemStoreDirectory(directoryPath);
+                var apiKey = path.ReadApiKey(_config);
+
+                if (!string.IsNullOrEmpty(apiKey))
+                {
+                    var created = OpenOrCreateChannel(apiKey, ApiKeyToName(apiKey));
+                    
+                    lock (_channelsSync)
+                    {
+                        _channels.Add(apiKey, created);
+                    }
+                }
+            }
         }
-        
+    }
+
+    string GetStorePath(string name)
+    {
+        return Path.Combine(_bufferPath, name);
+    }
+
+    public ForwardingChannel GetSeqCliConnectionChannel()
+    {
+        return _defaultChannel;
+    }
+
+    public ForwardingChannel GetApiKeyChannel(string apiKey)
+    {
         lock (_channelsSync)
         {
             if (_channels.TryGetValue(apiKey, out var channel))
             {
                 return channel;
             }
 
-            // Seq API keys begin with four identifying characters that aren't considered part of the
-            // confidential key. TODO: we could likely do better than this.
-            var name = apiKey[..4];
-            var created = OpenOrCreateChannel(apiKey, name);
+            var created = OpenOrCreateChannel(apiKey, ApiKeyToName(apiKey));
+            var store = new SystemStoreDirectory(GetStorePath(ApiKeyToName(apiKey)));
+            store.WriteApiKey(_config, apiKey);
             _channels.Add(apiKey, created);
             return created;
         }
     }
 
+    string ApiKeyToName(string apiKey)
+    {
+        // Seq API keys begin with four identifying characters that aren't considered part of the
+        // confidential key. TODO: we could likely do better than this.
+        return apiKey[..(Math.Min(apiKey.Length, 4))];
```

Comment:
On the next pass, if we fix/improve the directory naming scheme, we should include a min-length requirement and ignore payloads/keys below it. Seq's minimum is something like 16 chars IIRC.